docs/Config.md

```diff
@@ -0,0 +1,277 @@
+ 
+TESTS_TO_RUN: TestsToRun is a list of files which should be executed as part of a test suite
+ 
+SUPPRESS_SKIP_NOTIFICATIONS:  SuppressSkipNotifications suppresses the notifications of skipped tests
+ 
+CLEAN_RUNS: CleanRuns is the number of times the test-version is run before skipping.
+ 
+OPERATOR_SKIP:  OperatorSkip is a comma-delimited list of operator names to ignore health checks from. ex. "insights,telemetry"
+ 
+SKIP_CLUSTER_HEALTH_CHECKS: SkipClusterHealthChecks skips the cluster health checks. Useful when developing against a running cluster.
+ 
+METRICS_BUCKET: MetricsBucket is the bucket that metrics data will be uploaded to.
+ 
+SERVICE_ACCOUNT:  ServiceAccount defines what user the tests should run as. By default, osde2e uses system:admin
+ 
+ 
+ 
+ 
+Addon test related:-
+ 
+ADDON_IDS_AT_CREATION:  Comma separated list of IDs to create at cluster creation time
+ 
+ADDONS_IDS: Comma separated list of IDs to install after a cluster is created.
+ 
+ADDON_TEST_HARNESSES: TestHarnesses is a comma separated list of container images that will test the addon
+ 
+ADDON_TEST_USER:  TestUser is the OpenShift user that the tests will run as. If "%s" is detected in the TestUser string, it will evaluate that as the project namespace
+Example -> "system:serviceaccount:%s:dedicated-admin"
+Evaluated -> "system:serviceaccount:osde2e-abc123:dedicated-admin"
+ 
+ADDON_RUN_CLEANUP:  RunCleanup is a boolean to specify whether the testHarnesses should have a separate cleanup phase. This phase would run at the end of all e2e testing
+ 
+ADDON_CLEANUP_HARNESSES:  CleanupHarnesses is a comma separated list of container images that will clean up any artifacts created after test harnesses have run
+ 
+ 
+Prometheus related:-
+ 
+PROMETHEUS_ADDRESS: Address of the Prometheus instance to connect to.
+ 
+PROMETHEUS_BEARER_TOKEN:  Token needed for communicating with Prometheus.
+ 
+```
+
+## Command Line Flags for osde2e
+
+CLI flags that are commonly used include:
+ 
+For the test sub-command:
+--cluster-id: Existing OCM cluster ID to run tests against.
+--configs:  A comma separated list of built in configs to use. (stage, prod, e2e-suite, etc.)
+--custom-config: Custom config file for osde2e.
+--destroy-cluster: A flag to trigger cluster deletion after test completion.
+--environment: Cluster provider environment to use. (ocm, moa, etc.).
+--kube-config: Path to local Kube config for running tests against.
+--skip-health-check:  a flag to skip cluster health checks.
+--skip-tests: Skip any Ginkgo tests whose names match the regular expression.
+ 
+For the query sub-command:
+--output-format:  Output format for query results (json|prom). Defaults to json. (default "-")
+
+ 
+## Common config flag values
+
+The following are the values that can be plugged in for the --configs flag when running osde2e/osde2ectl. The values correspond to existing YAML files in the /configs folder:-
+
+1. OSD environment values:
```

Comment:
For each of these options, could you do a markdown table? I think it'll make the options easier to visually parse